src/wp-includes/functions.php

```diff
@@ -4266,6 +4266,40 @@ function wp_check_jsonp_callback( $callback ) {
 	return 0 === $illegal_char_count;
 }
 
+/**
+ * Reads and decodes a JSON file.
+ *
+ * @since 5.9.0
+ *
+ * @param string $filename Path to the JSON file.
+ * @param array  $options  {
+ *     Optional. Options to be used with `json_decode()`.
+ *
+ *     @type bool associative Optional. When `true`, JSON objects will be returned as associative arrays.
+ *                            When `false`, JSON objects will be returned as objects.
+ * }
+ *
+ * @return mixed Returns the value encoded in JSON in appropriate PHP type.
+ *               `null` is returned if the file is not found, or its content can't be decoded.
+ */
+function wp_json_file_decode( $filename, $options = array() ) {
+	$result = null;
+	if ( ! file_exists( $filename ) ) {
+		trigger_error( "File $filename doesn't exist!" );
+		return $result;
+	}
+
+	$options      = wp_parse_args( $options, array( 'associative' => false ) );
+	$decoded_file = json_decode( file_get_contents( $filename ), $options['associative'] );
```

Comment:
Might want to use `wp_normalize_path()` and perhaps `realpath` for the file name.